pymc3/sampling.py

```diff
@@ -316,9 +316,20 @@ def sample(draws=500, step=None, init='auto', n_init=200000, start=None, trace=N
     """
     model = modelcontext(model)
 
+    nuts_kwargs = kwargs.pop('nuts_kwargs', None)
+    if nuts_kwargs is not None:
```

Comment:
Instead of popping from the dict and then re-updating the dictionary, couldn't we just change this condition to `if 'nuts_kwargs' in kwargs` (and similarly for `step_kwargs`)?

Reply:
I want there to be a deprecation warning if they are present, so I'd have to peek at them anyhow.

Reply:
For example, before my update the `kwargs` should look like this:

`{'nuts_kwargs': {'target_accept': 0.99}, 'some_other_arg': 'foo'}`

since I've removed `nuts_kwargs` (and `step_kwargs`) from the argument list. I want it to look like this:

`{'target_accept': 0.99, 'some_other_arg': 'foo'}`